streams/src/main/java/org/apache/kafka/streams/kstream/KStream.java

```diff
@@ -14,641 +14,1705 @@
+     * This topic will be named "${applicationId}-XXX-repartition", where "applicationId" is user-specified in
+     * {@link org.apache.kafka.streams.StreamsConfig StreamsConfig} via parameter
+     * {@link org.apache.kafka.streams.StreamsConfig#APPLICATION_ID_CONFIG APPLICATION_ID_CONFIG}, "XXX" is an internal
+     * generated name, and "-repartition" is a fixed suffix.
+     * You can retrieve all generated internal topic names via {@link KafkaStreams#toString()}.
+     * <p>
+     * For this case, all data of this stream will be redistributed through the repartitioning topic by writing all
+     * records to and rereading all records from it, such that the resulting {@link KGroupedStream} is partitioned
+     * correctly on its key.
+     * If the last key changing operator changed the key type, it is recommended to use
+     * {@link #groupByKey(Serde, Serde)} instead.
      *
-     * @param processorSupplier         the supplier of {@link ProcessorSupplier} that generates {@link org.apache.kafka.streams.processor.Processor}
-     * @param stateStoreNames           the names of the state store used by the processor
+     * @return a {@link KGroupedStream} that contains the grouped records of the original {@link KStream}
+     * @see #groupBy(KeyValueMapper)
      */
-    void process(ProcessorSupplier<K, V> processorSupplier, String... stateStoreNames);
+    KGroupedStream<K, V> groupByKey();
 
     /**
-     * Combine element values of this stream with another {@link KStream}'s elements of the same key using windowed Inner Join.
-     * If a record key is null it will not included in the resulting {@link KStream}
-     * Both of the joining {@link KStream}s will be materialized in local state stores with auto-generated store names.
-     * Also a changelog topic named "${applicationId}-store name-changelog" will be automatically created
-     * in Kafka for each store for failure recovery, where "applicationID" is user-specified in the
-     * {@link org.apache.kafka.streams.StreamsConfig}.
-     *
-     * @param otherStream       the instance of {@link KStream} joined with this stream
-     * @param joiner            the instance of {@link ValueJoiner}
-     * @param windows           the specification of the {@link JoinWindows}
-     * @param keySerde          key serdes for materializing both streams,
-     *                          if not specified the default serdes defined in the configs will be used
-     * @param thisValueSerde    value serdes for materializing this stream,
-     *                          if not specified the default serdes defined in the configs will be used
-     * @param otherValueSerde   value serdes for materializing the other stream,
-     *                          if not specified the default serdes defined in the configs will be used
-     * @param <V1>              the value type of the other stream
-     * @param <R>               the value typ
```

Comment:
Perhaps say that the repartitioning will happen automatically, i.e., user doesn't need to do anything.